### ISSUE TITLE
Insufficent permission to catalog='system' when using dbt-databricks=1.8.7

### ISSUE THREAD

**Issue by VippsMe** [databricks/dbt-databricks#838]
### Describe the bug
A clear and concise description of what the bug is. What command did you run? What happened?

**Issue when using dbt-databricks==1.8.7**

- We just upgraded from dbt-databricks 1.5 to **1.8.7.**  This upgrade resulted in an error in the ci job. Downgraded to vs 1.7.17 and then it worked again.
- Error message in github action ci job:

[INSUFFICIENT_PERMISSIONS] Insufficient privileges:
    User does not have USE CATALOG on Catalog 'system'. SQLSTATE: 42501

### Steps To Reproduce
See above. Run dbt compile and it will give you the error message.

### Expected behavior
See above.

### Screenshots and log output
![image](https://github.com/user-attachments/assets/1c3cd2ff-65c0-453e-a678-2eea74bc87d5)

### System information
**The output of `dbt --version`:**
```
<output goes here>
```

**The operating system you're using:**

**The output of `python --version`:**

**Comment by benc-db** [comment]
This looks like your admin has removed your permission to the system catalog, as every user should have access to this table by default.  We can investigate a flag to allow you to reference UC information_schemas scoped by catalog, but we moved to using system catalog as the catalog-scoped information_schema views break if you rename a catalog, so it's kind of a no win situation.

**Comment by VippsMe** [comment]
> a

Its not because of our admin. Access to catalog 'system' has been the exact same all the time. 
So it is something about dbt-databricks adapter 1.8.XX that is trying to access catalog 'system' (and not our internal repo/code)

**Comment by benc-db** [comment]
@VippsMe yes, we changed to using system, but we changed because a.) it is more reliable than the individual catalog info_schemas because they are just views of system that can break if the catalog is renamed, and b.) everyone has access to system.information_schema by default.  So the reason I'm suggesting that your admin at some point removed your permission, is because you should have access, and if you don't, it's something that an admin can change.

**Comment by rheanes** [comment]
> It works in 1.7.17 but not in 1.8.7 and >
> 
> I work on a managed databricks workspace and my company will not allow me rights on the system_catalog.
> 
Did you ever figure out a work around for this?

**Comment by d474b3r9** [comment]
> > It works in 1.7.17 but not in 1.8.7 and >
> > I work on a managed databricks workspace and my company will not allow me rights on the system_catalog.
> 
> Did you ever figure out a work around for this?

You need those rights for your user, you have no other choices.

**Comment by benc-db** [comment]
We require select on system.information_schema.*.

**Comment by joaofonsecapromptly** [comment]
> We require select on system.information_schema.*.

Alright but is there any documentation on what you do with those tables? Why do you need it?

**Comment by benc-db** [comment]
To gather metadata about the objects so as to make decisions about what operations to take.  Some metadata is only currently available from tables in information_schema, such as certain constraint definitions.
